### PR TITLE
Check SONiC dependencies before installation.

### PR DESCRIPTION
**Pull request by liushilongbuaa**
#### Why I did it
SONiC related packages shouldn't be intalled from Pypi.
It is security compliance requirement.
#### How I did it
Check SONiC related packages when using setup.py.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

**Review by xumia — APPROVED** [review]

File: src/sonic-config-engine/setup.py [hunk 1]
```diff
@@ -50,6 +55,14 @@
         'sonic_yang_cfg_generator'
     ]
 
+for package in sonic_dependencies:
+    try:
+        pkg_resources.get_distribution(package)
+    except pkg_resources.DistributionNotFound:
+        print(package+" is not found!")
```
Comment (qiluo-msft) [inline src/sonic-config-engine/setup.py:62]: >print [](http://example.com/codeflow?start=8&length=5)

Print 2 error messages to stderr.

File: src/sonic-config-engine/setup.py [hunk 2]
```diff
@@ -50,6 +55,14 @@
...
+        print("Please build and install SONiC python wheels dependencies from github.com/sonic-net/sonic-buildimage")
```
Comment (qiluo-msft) [inline src/sonic-config-engine/setup.py:63]: >github.com/sonic-net/ [](http://example.com/codeflow?start=78&length=21)

You could remove url and only keep the repo name.

File: src/sonic-config-engine/setup.py [hunk 3]
```diff
@@ -24,6 +28,7 @@
         'sonic-yang-mgmt>=1.0',
         'sonic-yang-models>=1.0'
     ]
+    sonic_dependencies += ['sonic-yang-mgmt', 'sonic-yang-models']
```
[the remark this answers is not in the document]
Comment (liushilongbuaa) [reply src/sonic-config-engine/setup.py:31]: We can't generate from dependencies. Something like swsssdk didn't have prefix sonic-*.

File: src/sonic-config-engine/setup.py (hunk 1 above)
Comment (liushilongbuaa) [reply src/sonic-config-engine/setup.py]: It should be used by python2 and python3.
we can't use print. Use sys.stderr.write instead.

Comment (qiluo-msft) [reply src/sonic-config-engine/setup.py]: you can use print(msg, file=sys.stderr)

Comment (liushilongbuaa) [reply src/sonic-config-engine/setup.py]: python2 doesn't support it.

File: src/sonic-config-engine/setup.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (qiluo-msft) [reply src/sonic-config-engine/setup.py]: sonic_dependencies could be ground truth, including the package names and version comparison.

1. install_requires could take the package names and version comparison. No need to repeat each package.
2. get_distribution check could take the package names, and fetch local versions, and compare with expected condition.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**